core/DataAccess/TableMetadata.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\DataAccess;
+
+use Piwik\Db;
+
+/**
+ * Data Access Object that can be used to get metadata information about
+ * the MySQL tables Piwik uses.
+ */
+class TableMetadata
```

Comment:
Yeah, my thoughts are that getColumns is pretty generic, but getIdActionColumnNames is specific to certain use-cases/functionality. Ideally, I think all relational stuff would be in its own plugin (ie, plugins/MySQL or plugins/Relational, etc.). This way it wouldn't need to be in core, but Piwik is pretty far away from the possibility of defining a data access layer in a plugin.

FYI, I only moved it in response to @tsteur last comment about making as much re-usable as possible.